integration/compare_sunbird_with_glpi.py

```diff
@@ -0,0 +1,357 @@
+    )
+    parser.add_argument(
+        "-u",
+        "--username",
+        type=str,
+        required=True,
+        help="Username of Sunbird service account",
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        type=str,
+        required=True,
+        help="Password of Sunbird service account",
+    )
+    parser.add_argument(
+        "-s", "--sunbird_url", type=str, required=True, help="URL of Sunbird Instance"
+    )
+    parser.add_argument(
+        "-r",
+        "--recipient",
+        type=str,
+        help="Recipient of email",
+    )
+    parser.add_argument(
+        "-S",
+        "--sender",
+        type=str,
+        help="Sender of email. Address doesn't necessarily need to be valid.",
+    )
+    parser.add_argument(
+        "-e",
+        "--email_server",
+        type=str,
+        help="Server used to send email",
+    )
+    args = parser.parse_args()
+
+    # Process General Config
+    with open(args.general_config, "r") as config_path:
+        config_map = yaml.safe_load(config_path)
+
+    user_token = args.token
+    ip = args.ip
+    global concise
+    concise = args.yaml
+    no_verify = args.no_verify
+    sunbird_username = args.username
+    sunbird_password = args.password
+    sunbird_url = validate_url(args.sunbird_url)
+    email_recipient = args.recipient
+    email_sender = args.sender
+    email_server = args.email_server
+    enable_email = check_email_parameters(email_recipient, email_sender, email_server)
+
+    urls = UrlInitialization(ip)
+
+    with SessionHandler(user_token, urls, no_verify) as session:
+        headers = {"Accept": "application/json", "Content-Type": "application/json"}
+
+        sunbird_machines = get_sunbird_machines(
+            config_map, headers, sunbird_url, sunbird_username, sunbird_password
+        )
+
+        glpi_machines = get_glpi_machines(session, urls)
+
+        glpi_machines_in_sunbird = get_glpi_machines_in_sunbird(
+            glpi_machines, headers, sunbird_url, sunbird_username, sunbird_password
+        )
+
+        # Compare the two dictionaries
+        sunbird_only = set(sunbird_machines.keys()) - set(glpi_machines.keys())
+        glpi_only = set(glpi_machines.keys()) - set(glpi_machines_in_sunbird.keys())
+
+        # Output the results
+        sunbird_only_list = list(sunbird_only)
+        glpi_only_list = list(glpi_only)
+
+        output = (
+            f"There are {len(sunbird_only_list)} computers in Sunbird but not in GLPI:"
+            "\n"
+            f"{sunbird_only_list}"
+            "\n\n"
+            f"There are {len(glpi_only_list)} computers in GLPI but not in Sunbird:"
+            "\n"
+            f"{glpi_only_list}"
+        )
+
+        if enable_email:
+            send_email(output, email_recipient, email_sender, email_server)
+        print(output)
+
+    if not concise:
+        print_final_help()
+
+
+def get_sunbird_machines(
```

Comment:
Can't this method make "get_glpi_machines_in_sunbird" redundant? Seems like either taking in a flag or checking a dict for serial number vs location? If this is more effort than it's worth then feel free to ignore it

Reply:
I think I prefer to actually keep them separate, since they are kind of doing different things:

`def get_sunbird_machines()` searches across ordered pairs of Locations and Cabinets (cmbLocation and cmbCabinet in Sunbird), and then compiles all of the information into a single dictionary. 

On the other hand, `def get_glpi_machines_in_sunbird` searches across a list of serial numbers (tiSerialNumber in Sunbird).

Reply:
Alright, sounds good